functional-samples/ai.gemini-on-device-summarization/sidepanel/index.js

```diff
@@ -55,39 +78,30 @@ async function generateSummary(text) {
   }
 }
 
-async function createSummarizationSession(downloadProgressCallback) {
+async function createSummarizer(config, downloadProgressCallback) {
   if (!window.ai || !window.ai.summarizer) {
     throw new Error('AI Summarization is not supported in this browser');
   }
   const canSummarize = await window.ai.summarizer.capabilities();
   if (canSummarize.available === 'no') {
-    throw new Error('AI Summarization is not availabe');
+    throw new Error('AI Summarization is not supported');
   }
-
-  const summarizationSession = await window.ai.summarizer.create();
+  const summarizationSession = await window.ai.summarizer.create(
```

Comment:
```suggestion
  const summarizationSession = await self.ai.summarizer.create(
```

Shorter and worker-compatible.